src/renderers/shared/fiber/ReactFiberCompleteWork.js

```diff
@@ -132,6 +130,17 @@ module.exports = function<T, P, I, C>(config : HostConfig<T, P, I, C>) {
         return null;
       case ClassComponent:
         transferOutput(workInProgress.child, workInProgress);
+        // Don't use the state queue to compute the memoized state. We already
+        // merged it and assigned it to the instance. Transfer it from there.
+        const state = workInProgress.stateNode.state;
+        workInProgress.memoizedState = state;
+        // Transfer update queue to callbackList field so callbacks can be
+        // called during commit phase.
+        workInProgress.callbackList = workInProgress.updateQueue;
+        if (current) {
+          current.callbackList = workInProgress.callbackList;
```

Comment:
Hmm I think I see what you mean. So the only place we should ever reset the queue / callback list is after the changes are committed?